source/s3_default_meta_request.c

```diff
@@ -143,7 +158,14 @@ static bool s_s3_meta_request_default_update(
                     goto has_work_remaining;
                 }
 
-                request = aws_s3_request_new(meta_request, 0, 1, AWS_S3_REQUEST_FLAG_RECORD_RESPONSE_HEADERS);
+                request = aws_s3_request_new(
+                    meta_request,
+                    0 /*request_tag*/,
+                    AWS_S3_REQUEST_TYPE_DEFAULT,
+                    meta_request_default->operation_name ? aws_string_c_str(meta_request_default->operation_name)
+                                                         : NULL,
```

Comment:
debatable: will it be better to have a default string instead of null if nothing was set? 
The reason I think is:
- Without setting a operation name in this case is not an ERROR, using a NULL more like an error to me.
- The TYPE enum we have a default value for the type, but the operation name does not.

Reply:
yeah, it's super debatable whether to have "" or NULL, and whether that differs based on if you're storing `const char *` or `aws_string *`.

Anyway, these are `aws_string *` now, and I'm still using NULL for defaults. because allocating an aws_string, just to store "" is too weird for me